### ISSUE TITLE
Tuple conversion error & converting error 

### ISSUE THREAD

**Issue by fxwx23** [ReactKit/SwiftTask#46]
I'm using ReactKit (Swift/2.0 branch) through cocoapods in my xcode project.
I'm getting below error while trying to build. (**_Xcode7.0 beta6, Swift2.0**_)

> Cannot express tuple conversion '(error: Error?, isCancelled: Bool)' (aka '(error: Optional<Error>, isCancelled: Bool)') to '(error: Error??, isCancelled: Bool)' (aka '(error: Optional<Optional<Error>>, isCancelled: Bool)')

![2015-08-31 17 13 41](https://cloud.githubusercontent.com/assets/7478439/9576038/2e425d82-500f-11e5-9d38-446a5e57587c.png)

I've tried to fix this like below

``` Swift
public func success<Progress2, Value2, Error2, C: Canceller>(inout canceller: C?, _ successClosure: Value -> Task<Progress2, Value2, Error2>) -> Task<Progress2, Value2, Error>
    {
        return Task<Progress2, Value2, Error> { [unowned self] newMachine, progress, fulfill, _reject, configure in

            let selfMachine = self._machine

            // NOTE: using `self._then()` + `selfMachine` instead of `self.then()` will reduce Task allocation
            self._then(&canceller) {
                if let value = selfMachine.value.rawValue {
                    let innerTask = successClosure(value)
                    _bindInnerTask(innerTask, newMachine: newMachine, progress: progress, fulfill: fulfill, _reject: _reject, configure: configure)
                }
                else if let errorInfo = selfMachine.errorInfo.rawValue {
                    _reject(ErrorInfo(errorInfo.error, errorInfo.isCancelled))
                }
            }
        }.name("\(self.name)-success")
    }
```

however, I'm still have below error. I'm not sure how can I change return expression. 

> Cannot convert return expression of type 'Task<Progress2, Value2, Error?>' (aka 'Task<Progress2, Value2, Optional<Error>>') to return type 'Task<Progress2, Value2, Error>'

Thank you in beforehand.

**Comment by inamiy** [comment]
I think it's already fixed in 52b976d106da7417715cab6a122ffbaae6913424.
If you are using ReactKit, please make sure to specify SwiftTask's `swift/2.0` branch as well and `pod update`.

**Comment by fxwx23** [comment]
thanks.
I've solved this now like you said.
